src/location-distance-utils.c

```diff
@@ -21,31 +21,33 @@
 
 #include "location-distance-utils.h"
 
-static double d2r(double deg)
-{
-	return (deg * M_PI / 180);
-}
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+// scaling factor from degrees to meters at equator
+// == DEG_TO_RAD * radius of earth
+static float LOCATION_SCALING_FACTOR = 111318.84502145034f;
+
+#define MAX(a, b) (((a) > (b)) ? (a) : (b))
+
+#define D2R (M_PI / 180.0f)
+// #define R2D (180.0f / M_PI)
 
-static double r2d(double rad)
+// compensate for shrinking longitude towards poles
+float lng_scale(double lat)
 {
-	return (rad * 180 / M_PI);
+	float scale = cosf(lat * D2R);
+	return MAX(scale, 0.01f);
 }
 
+// return distance in meters between two locations
 double location_distance_between(double latitude_s,
-		double longitude_s,
-		double latitude_f,
-		double longitude_f)
+								 double longitude_s,
+								 double latitude_f,
+								 double longitude_f)
 {
-	double theta, dist;
-
-	theta = longitude_s - longitude_f;
-	dist = sin(d2r(latitude_s)) * sin(d2r(latitude_f)) \
-		+ cos(d2r(latitude_s)) * cos(d2r(latitude_f)) \
-		* cos(d2r(theta));
-	dist = acos(dist);
-	dist = r2d(dist);
-	dist = dist * 60 * 1.1515;
-	dist = dist * 1.609344; /* kilometers */
-
-	return dist;
+	float dlat = (float)(latitude_f - latitude_s);
```

Comment:
I haven't verified the maths here (it was too long ago I was playing with trigonometric functions when I was student), so could you provide some evidence that either original code does it wrongly or the replacement calculates the same values given same input values.

Reply:
The original code uses `Spherical law of cosines` which gives good results down to distances as small as few meters,
what is used here is `Equirectangular approximation` which is accurate for small distances only but performs well.
of course there's  `Haversine` which is accurate for all king of distances but considered relatively slow.
I could add `Haversine` calculations, what do you think?
here's a [link](https://www.movable-type.co.uk/scripts/latlong.html) talks about all three laws